common/include/pcl/conversions.h

```diff
@@ -193,12 +196,13 @@ namespace pcl
       // Should usually be able to copy all rows at once
       if (msg.row_step == cloud_row_step)
       {
-        memcpy (cloud_data, msg_data, msg.data.size ());
+        std::copy(msg.data.begin(), msg.data.end(), cloud_data);
       }
       else
       {
-        for (std::uint32_t i = 0; i < msg.height; ++i, cloud_data += cloud_row_step, msg_data += msg.row_step)
-          memcpy (cloud_data, msg_data, cloud_row_step);
+        for (std::uint32_t i = 0; i < msg.height; ++i, cloud_data += cloud_row_step, msg_data += msg.row_step) {
+          std::copy(msg.data.begin(), msg.data.begin() + cloud_row_step, cloud_data);
```

Comment:
It feels like the logic changed. Could you please check this again?

Reply:
Revised